### PR TITLE
[chore] Update Prometheus to v0.40.1

### LINKED ISSUE #16299: updating prometheus library causes problems
### Component(s)

receiver/prometheus

### Describe the issue you're reporting

The suggested [dependabot update](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16289) to the prometheus library causes problems:

```
$ go get github.com/prometheus/prometheus@v0.40.1 && make tidy && make test
go test -race -v -timeout 300s --tags="" ./...
# github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal
internal/appendable.go:74:9: cannot use newTransaction(ctx, o.metricAdjuster, o.sink, o.externalLabels, o.settings, o.obsrecv) (value of type *transaction) as type storage.Appender in return statement:
        *transaction does not implement storage.Appender (missing AppendHistogram method)
FAIL    github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver [build failed]
FAIL    github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal [build failed]
FAIL
make: *** [test] Error 2
```

### PR DESCRIPTION
**Pull request by Aneurysm9**
Signed-off-by: Anthony J Mirabella <a9@aneurysm9.com>

Fixes #16299.

### CODE REVIEW

File: receiver/prometheusreceiver/internal/transaction.go [hunk 1]
```diff
@@ -179,6 +180,10 @@ func (t *transaction) AppendExemplar(ref storage.SeriesRef, l labels.Labels, e e
 	return 0, nil
 }
 
+func (t *transaction) AppendHistogram(ref storage.SeriesRef, l labels.Labels, ts int64, h *histogram.Histogram) (storage.SeriesRef, error) {
+	return 0, fmt.Errorf("Not Implemented")
```
Comment (Aneurysm9) [inline receiver/prometheusreceiver/internal/transaction.go:184]: Not sure whether this should return an error or `nil`.  Open to suggestions.

**Comment by Aneurysm9** [comment]
[Apparently](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3484190004/jobs/5828470852#step:5:458), we can't update until `weaveworks/common` fixes a breaking change in a prometheus exporter helper.  It would be really, really nice if Prometheus stopped breaking BC in what are supposed to be minor releases.

File: receiver/prometheusreceiver/internal/transaction.go (hunk 1 above)
Comment (dashpole) [reply receiver/prometheusreceiver/internal/transaction.go:184]: IIRC, a failure to append one series in a scrape fails the entire scrape.  Do we know if that is the case here?   Seems safest to return nil.  We could debug log if we wanted to be helpful

Before we supported exemplars, we would return 0, nil in AppendExemplar: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/f21f89eeb154975bf23fc74fabf0163e7626da28/receiver/prometheusreceiver/internal/otlp_transaction.go#L103

**Review by dashpole — APPROVED** [review]

**Comment by bboreham** [comment]
Possibly https://github.com/weaveworks/common/pull/262 will help.

**Review by codeboten — COMMENTED** [review]
@Aneurysm9 can this be moved forward with the common package update?

**Comment by bogdandrutu** [comment]
@codeboten here is a clean/ready to merge PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16734

**Comment by codeboten** [comment]
> @codeboten here is a clean/ready to merge PR #16734

Hah, i was just updating this PR with the 0.40.5 update

**Review by codeboten — COMMENTED** [review]
Superseeded by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16734

**Comment by codeboten** [comment]
Super

**Pull request closed without merging**